### PR TITLE
Test svcat gracefully handles the ns-broker flag being disabled

### PR DESCRIPTION
**Pull request by carolynvs**
`svcat` will need to gracefully handle interacting with servers that have the feature flag for namespaced broker resources disabled. 

* The list commands that have been modified to retrieve both cluster and ns-scoped resources will ignore "resource not found" errors and return just the cluster resources. 
* FYI, as we add support for retrieving a single ns-scoped resource, like `svcat describe broker --namespace NS NAME`, I believe that should be a real error. Since that wasn't a supported command before. When we add support for those, we'll add test cases to ensure the proper error behavior.

We'll probably want to keep this around for at least until the feature flag is removed, perhaps longer so that people don't need to be careful about which client version they use with a particular version of the api server.

### CODE REVIEW

File: cmd/svcat/svcat_test.go [hunk 1]
```diff
@@ -41,17 +41,100 @@ import (
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8stesting "k8s.io/client-go/testing"
 )
 
 var catalogRequestRegex = regexp.MustCompile("/apis/servicecatalog.k8s.io/v1beta1/(.*)")
 var coreRequestRegex = regexp.MustCompile("/api/v1/(.*)")
 
+// Verify that svcat gracefully handles when the namespaced broker feature flag is disabled
+// TODO: Once we take Namespaced brokers out from behind the feature flag, this test won't be necessary
```
Comment (jeremyrickard) [inline cmd/svcat/svcat_test.go]: probably once we get it out from the feature flag + some number of versions?

Comment (carolynvs) [reply cmd/svcat/svcat_test.go]: Yeah I think you are right, I'm just not sure how long yet. Maybe until 1.0 honestly...

**Pull request merged**